tests/libraries/token/test_stake_token.py

```diff
@@ -0,0 +1,325 @@
+    mockReportingWindow.setUniverse(mockUniverse.address)
+    mockReportingWindow.setIsContainerForMarket(True)
+    mockReportingWindow.setNoteReportingGasPrice(True)
+    # wire up mock market
+    mockMarket = tokenFixture.contracts['MockMarket']
+    mockMarket.setNumberOfOutcomes(2)
+    mockMarket.setNumTicks(numTicks)
+    mockMarket.setUniverse(mockUniverse.address)
+    mockMarket.setReportingWindow(mockReportingWindow.address)
+    mockMarket.setIsContainerForStakeToken(True)
+    mockMarket.setDerivePayoutDistributionHash(stringToBytes("1"))
+
+    stakeToken = tokenFixture.upload('../source/contracts/reporting/StakeToken.sol', 'stakeToken')
+    assert stakeToken.initialize(mockMarket.address, [0, numTicks], False)       
+    
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].PRE_REPORTING()) 
+    with raises(Exception, message="market can't call trust buy unless market state is round 1 or round 2"):
+        mockMarket.callStakeTokenTrustedBuy(stakeToken, tester.a1, 1)
+
+    # call stake token successfully
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].ROUND1_REPORTING()) 
+    assert mockMarket.callStakeTokenTrustedBuy(stakeToken.address, tester.a1, 1)
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].ROUND2_REPORTING()) 
+    assert mockMarket.callStakeTokenTrustedBuy(stakeToken.address, tester.a1, 1)
+
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].LAST_DISPUTE()) 
+    with raises(Exception, message="market can't call trust buy unless market state is round 1 or round 2"):
+        mockMarket.callStakeTokenTrustedBuy(stakeToken, tester.a1, 1)
+
+@fixture(scope="session")
+def localSnapshot(fixture, kitchenSinkSnapshot):
+    fixture.resetToSnapshot(kitchenSinkSnapshot)
+    fixture.uploadAndAddToController('solidity_test_helpers/MockMarket.sol')
+    fixture.uploadAndAddToController('solidity_test_helpers/MockReportingWindow.sol')
+    fixture.uploadAndAddToController('solidity_test_helpers/MockUniverse.sol')
+    fixture.uploadAndAddToController('solidity_test_helpers/MockReputationToken.sol')
+    assert fixture.contracts['MockReputationToken']
+    universe = ABIContract(fixture.chain, kitchenSinkSnapshot['universe'].translator, kitchenSinkSnapshot['universe'].address)
+    market = ABIContract(fixture.chain, kitchenSinkSnapshot['binaryMarket'].translator, kitchenSinkSnapshot['binaryMarket'].address)
+    return initializeReportingFixture(fixture, universe, market)
+
+@fixture
+def universe(fixture, kitchenSinkSnapshot):
+    universe = ABIContract(fixture.chain, kitchenSinkSnapshot['universe'].translator, kitchenSinkSnapshot['universe'].address)
+    return universe
+
+@fixture
+def binaryMarket(fixture, kitchenSinkSnapshot):
+    market = ABIContract(fixture.chain, kitchenSinkSnapshot['binaryMarket'].translator, kitchenSinkSnapshot['binaryMarket'].address)
+    return market
+
+@fixture
+def cash(fixture, kitchenSinkSnapshot):
```

Comment:
These 4 fixtures (universe, binaryMarket, categorigcalMarket, cash) should take in `tokenFixture` as the first parameter, not fixture.  This will guarantee that they are called _after_ `tokenFixture`, which is necessary to ensure that the ABI is mapped correctly after the fixture was last reset.